rag/nlp/__init__.py

```diff
@@ -471,7 +471,8 @@ def add_chunk(t, pos):
         tnum = num_tokens_from_string(t)
         if tnum < 8:
             pos = ""
-        if tk_nums[-1] > chunk_token_num:
+        # Ensure that the length of the merged chunk does not exceed chunk_token_num  
+        if tk_nums[-1] + tnum > chunk_token_num:

```

Comment:
```suggestion
        if tk_nums[-1] > chunk_token_num:

```
It's very easy to split title from its description.

**TITLE**
Description .... 128 and more ....